contracts/facets/GBMFacet.sol

```diff
@@ -212,7 +217,42 @@ contract GBMFacet is IGBM, IERC1155TokenReceiver, IERC721TokenReceiver, Modifier
         bytes4 _tokenKind,
         bool _useInitiator
     ) public onlyOwner {
-        modifyAnAuctionToken(_tokenContract, _tokenId, _tokenKind, _useInitiator, 0, false);
+        modifyAnAuctionToken(_tokenContract, _tokenId, _tokenKind, _useInitiator, 0, 0, 1, 0);
+    }
+	
+	/// @notice Register an auction token and emit the relevant AuctionInitialized & AuctionStartTimeUpdated events
+    /// Throw if the token owner is not the GBM smart contract/supply of auctionned 1155 token is insufficient
+    /// @param _contractID Id of the token contract the auctionned token belong to
+    /// @param _tokenId The token ID of the token being auctionned
+    /// @param _tokenKind either bytes4(keccak256("ERC721")) or bytes4(keccak256("ERC1155"))
+    /// @param _auctionPresetID Choosen preset for the auction
+    function registerAnAuctionTokenSecondaryMarket(
+        uint256 _contractID,
+        uint256 _tokenId,
+        bytes4 _tokenKind,
+        uint256 _tokenAmount,
+        uint256 _auctionPresetID
+    ) public {
+        address _tokenContract = s.secondaryMarketTokenContract[_contractID];
+        modifyAnAuctionToken(_tokenContract, _tokenId, _tokenKind, false, 0, 0, _tokenAmount, _auctionPresetID);
+		uint256 localIndex;
+		if (_tokenKind == bytes4(keccak256("ERC721"))){
+            IERC721(_tokenContract).safeTransferFrom(msg.sender, address(this), _tokenId);
+			localIndex = s.erc721TokensIndex[_tokenContract][_tokenId] -1;
+			s.auctionSeller[s.auction721Mapping[_tokenContract][_tokenId][localIndex]] = msg.sender;
+		} else {
+            IERC1155(_tokenContract).safeTransferFrom(msg.sender, address(this), _tokenId, _tokenAmount, "");
+			localIndex =  s.erc1155TokensIndex[_tokenContract][_tokenId] -1;
```

Comment:
underflows for new auctions....consider removing `-1`

Reply:
since they are new auctions, default value will be 0... a subctraction will lead to an underflow